### PR TITLE
Support KILL <processID> command.

### LINKED ISSUE #20299: support KILL <processID> to cancel running SQL
## Feature Request

**For English only**, other languages will not accept.

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot make decision by current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Describe the feature you would like.

`SHOW PROCESSLIST` will list current running SQL, but not support `KILL processID` command to cancel running SQL.

### PR DESCRIPTION
**Pull request by ICannerxy**
Fixes #20299.

Changes proposed in this pull request:
- Support KILL <processID>

### CODE REVIEW

**Comment by tuichenchuxin** [comment]
Looks Greate! Can you briefly the design principles?

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dal/KillStatementContext.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+package org.apache.shardingsphere.infra.binder.statement.dal;
+
+import org.apache.shardingsphere.infra.binder.statement.CommonSQLStatementContext;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.dal.MySQLKillStatement;
+
+/**
+ * Kill process statement.
+ */
+public class KillStatementContext extends CommonSQLStatementContext<MySQLKillStatement> {
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dal/KillStatementContext.java:26]: Please add final for KillStatementContext.

File: shardingsphere-infra/shardingsphere-infra-executor/src/main/java/org/apache/shardingsphere/infra/executor/kernel/model/ExecutionGroupContext.java [hunk 2]
```diff
@@ -38,7 +38,7 @@
     
     private final Collection<ExecutionGroup<T>> inputGroups;
     
-    private final String executionID = new UUID(ThreadLocalRandom.current().nextLong(), ThreadLocalRandom.current().nextLong()).toString();
+    private final String executionID = new UUID(ThreadLocalRandom.current().nextLong(), ThreadLocalRandom.current().nextLong()).toString().replaceAll("-", "");
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-executor/src/main/java/org/apache/shardingsphere/infra/executor/kernel/model/ExecutionGroupContext.java:41]: Why add `.replaceAll("-", "")` for `executionID`?

File: shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/process/GovernanceExecuteProcessReporter.java [hunk 3]
```diff
@@ -60,10 +67,30 @@ public void report(final String executionID, final ExecuteProcessConstants const
...
     
     @Override
     public void reportClean(final String executionID) {
         ShowProcessListManager.getInstance().removeProcessContext(executionID);
     }
+    
+    private List<Statement> collectProcessStatement(final ExecutionGroupContext<? extends SQLExecutionUnit> executionGroupContext) {
+        List<Statement> statements = new ArrayList<>();
+        if (null == executionGroupContext.getInputGroups()) {
+            return statements;
```
Comment (strongduanmu) [inline shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/process/GovernanceExecuteProcessReporter.java:81]: Please rename statements to result.

File: shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/process/event/KillProcessIdRequestEvent.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Kill process id request event.
+ */
+@RequiredArgsConstructor
+@Getter
+public class KillProcessIdRequestEvent {
```
Comment (strongduanmu) [inline shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/process/event/KillProcessIdRequestEvent.java:28]: Please add final for KillProcessIdRequestEvent.

File: shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/status/compute/event/KillProcessIdUnitCompleteEvent.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.shardingsphere.mode.manager.cluster.coordinator.registry.GovernanceEvent;
+
+/**
+ * Kill processID unit complete event.
+ */
+@RequiredArgsConstructor
+@Getter
+public class KillProcessIdUnitCompleteEvent implements GovernanceEvent {
```
Comment (strongduanmu) [inline shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/status/compute/event/KillProcessIdUnitCompleteEvent.java:29]: Please add final for KillProcessIdUnitCompleteEvent.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by ICannerxy** [comment]
> Looks Greate! Can you briefly the design principles? @tuichenchuxin 

> `KILL <processID>` is depends on the improvement of `show processlist` , kill the processId of the process list.
design principles: We don't know the processID belongs to which `shardingSphere` instance, so when `shardingSphere` receives the Kill Command, it needs to notify each node, each node determines whether it currently has this ProcessId. If yes, find the statement connecting to the database according to the ProcessId, and then call statement.cancel to cancel the SQL task. 
The details are as follows:
1. DDLStatement and DMLStatement need save processID to Statement Map
2. Use the `zookeeper` node to implement wait for notification mechanism

**Comment by ICannerxy** [comment]
> Looks Greate! Can you briefly the design principles? @tuichenchuxin 

> `KILL <processID>` is depends on the improvement of `show processlist` , kill the processId of the process list.
design principles: We don't know the processID belongs to which `shardingSphere` instance, so when `shardingSphere` receives the Kill Command, it needs to notify each node, each node determines whether it currently has this ProcessId. If yes, find the statement connecting to the database according to the ProcessId, and then call statement.cancel to cancel the SQL task. 
The details are as follows:
1. DDLStatement and DMLStatement need save processID to Statement Map
2. Use the `zookeeper` node to implement wait for notification mechanism

File: shardingsphere-infra/shardingsphere-infra-executor/src/main/java/org/apache/shardingsphere/infra/executor/kernel/model/ExecutionGroupContext.java (hunk 2 above)
Comment (ICannerxy) [reply shardingsphere-infra/shardingsphere-infra-executor/src/main/java/org/apache/shardingsphere/infra/executor/kernel/model/ExecutionGroupContext.java:41]: DALStatement.g4 use `KILL (CONNECTION | QUERY)? IDENTIFIER_`, `IDENTIFIER_` not contains `-`

**Comment by tuichenchuxin** [comment]
> > Looks Greate! Can you briefly the design principles? @tuichenchuxin
> 
> > `KILL <processID>` is depends on the improvement of `show processlist` , kill the processId of the process list.
> > design principles: We don't know the processID belongs to which `shardingSphere` instance, so when `shardingSphere` receives the Kill Command, it needs to notify each node, each node determines whether it currently has this ProcessId. If yes, find the statement connecting to the database according to the ProcessId, and then call statement.cancel to cancel the SQL task.
> > The details are as follows:
> 
> 1. DDLStatement and DMLStatement need save processID to Statement Map
> 2. Use the `zookeeper` node to implement wait for notification mechanism

Got it. LGTM. This pr is only for cluster?

**Comment by ICannerxy** [comment]
> 

Yes

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**